common/persistence/Pinot/pinotVisibilityStore.go

```diff
@@ -656,13 +676,61 @@ func getListWorkflowExecutionsByQueryQuery(tableName string, request *p.ListWork
 		query.concatSorter(requestQuery)
 		query.addLimits(request.PageSize)
 	} else {
-		query.filters.addQuery(request.Query)
+		queryList := strings.Split(requestQuery, "and") //TODO: CAREFUL!!! do we need to check others like "AND" & "And" ?
+
+		for _, element := range queryList {
+			element := strings.TrimSpace(element)
+			pair := strings.Split(element, " ")
+			key := pair[0]
+
+			// case: when order by query also passed in
+			if strings.ToLower(key) == "order" && strings.ToLower(pair[1]) == "by" {
+				query.concatSorter(element)
+				continue
+			}
+
+			if pinotContainsKey(key) {
+				query.filters.addQuery(element)
+				continue
+			}
+
+			valType, ok := validMap[key]
+
+			if ok {
```

Comment:
nit: this can be written as:
`if valType, ok := validMap[key]; ok {
} else {
}`